packages/flutter_tools/test/general.shard/build_system/build_system_test.dart

```diff
@@ -375,6 +375,31 @@ void main() {
     expect(environmentA.buildDir.path, isNot(environmentB.buildDir.path));
   });
 
+  testWithoutContext('Additional inputs do not change the build configuration',  () async {
```

Comment:
Is there a test that changing a file provided by -i dirties the targets that reference it?

Reply:
It would be up to the target to include any in its depfile manually. I didn't add another test because there are no usages yet.

My gut says this is probably safe enough, but we could make it more automatic. Target gets an optional field `additionalInputs`. Then the build system automatically appends any additional inputs to the target's dependencies. WDUT?

Reply:
I think some test that the -i inputs get plumbed where they need to go would be good. If that's easier to test after some target is using one, then that sgtm.

Reply:
Will do then